Inventory - Internal Transfers

rithviknishad opened the pull request.

rithviknishad committed:
internal transfers layout

rithviknishad committed:
Requests Raised page

rithviknishad committed:
add comment

abhimanyurajeesh commented on src/pages/Facility/services/inventory/internalTransfer/ToReceive.tsx:
```diff
@@ -0,0 +1,151 @@
+  | "abandoned"
+  | "entered_in_error";
+
+function EmptyState() {
```
src/components/ui/empty-state.tsx

abhimanyurajeesh committed:
Add ToDispatch component and related table for internal transfer requests

- Introduced the ToDispatch component for managing internal transfer requests.
- Added ToDispatchSupplyRequestTable for displaying supply requests with filtering options.
- Updated en.json with new translation keys for dispatch requests and requests to dispatch.
- Integrated new routes in LocationLayout for navigating to the ToDispatch component.

yash-learner committed:
Add tables for ToDispatch tabs

yash-learner committed:
Use SDT in ToReceive tabs

amjithtitus09 committed:
Inventory - Dispatch Items screen

yash-learner committed:
Add delivery receive form (#12630)

yash-learner committed:
Make more changes
- Add alert dialog
- Move the entered in error to dropdown
- Some tiny changes
- Add close button
- ...

rithviknishad committed:
raise stock request

abhimanyurajeesh committed:
Enhance Purchase Orders component

- Added new translation keys to en.json for pending purchase orders and no purchase orders found description.
- Refactored PurchaseOrders component to include tabs for filtering by status and priority.
- Implemented responsive filters using Popover and Command components for item and priority selection.
- Updated query logic to handle tab changes and improved data fetching with TanStack Query.

Jacobjeevan committed:
purchase order table

abhimanyurajeesh committed:
Enhance Table styling and layout

- Adjusted TableHeader text size for better readability.
- Modified TableCell widths for improved layout consistency.
- Updated button styling for better visual alignment and responsiveness.
- Enhanced quantity display with bold styling for emphasis.

abhimanyurajeesh committed:
Refactor button and table components for improved styling and responsiveness

- Updated button sizes and styles for better alignment and consistency.
- Enhanced SupplyDeliveryTable and ToDispatchSupplyRequestTable with improved layout and text sizing.
- Added responsive design elements to table components for better usability on various screen sizes.
- Introduced EmptyState and TableSkeleton components for better loading and no data states in ToReceiveSupplyRequestTable.

abhimanyurajeesh committed:
Enhance table components with improved styling and responsiveness

- Updated TableBody components to include text-base for better readability.
- Adjusted button sizes from small to medium for improved usability.
- Replaced CareIcon with Eye icon for better visual representation in PurchaseOrderTable.
- Enhanced ServiceRequestTable with hover effects and improved layout consistency.

yash-learner committed:
Add more alert dialogs and minor changes

yash-learner committed:
Make deliver_from required and fix route so that

sidebar gets highlighted

yash-learner committed:
Fix the RaiseStockRequest form

- Mostly making the deliver_from as required caused the issue
- Now copying of shared data happens during on submit

yash-learner committed:
Remove condition for mark all as received checkbox

yash-learner committed:
Make some layout changes based on Vinu's suggestion

- Do not render side nav for receive form
- Add fixed max width
- other tiny style changes

yash-learner committed:
Use mutateAsync and do all invalidations

yash-learner committed:
Show unit from item definitinal dosage

rithviknishad committed:
Merge branch 'labs-v2' into inventory-internal-transfers

yash-learner committed:
Render the disptach form without the sidebar ...

- Add max-w
- Improve styling at some places

yash-learner committed:
Update the url

yash-learner committed:
Update the urls

Jacobjeevan committed:
receive items buttons

rithviknishad committed:
update raise request table and fixed minor bugs

abhimanyurajeesh committed:
Merge remote-tracking branch 'upstream/labs-v2' into inventory-internal-transfers

abhimanyurajeesh committed:
Enhance UI and localization for inventory and pharmacy components

- Added new translations for "no_requests_found" and its description in en.json.
- Refactored button styles for improved consistency.
- Updated InventoryList, SupplyDeliveryTable, ToDispatchSupplyRequestTable, MedicationDispenseHistory, MedicationRequestList, and ServiceRequestList components to use a unified Table component structure.
- Improved responsiveness and styling across various tables.

yash-learner committed:
Add comon BackButton component

yash-learner committed:
WIP: Add request detail page

yash-learner committed:
Add commom confirm dialog

yash-learner committed:
WIP: Use confirm dialog inside the SRD

yash-learner committed:
Some tiny changes

- Add action can't be undeone for cancel, error
- Add active for draft status in dropdown
- Some styles changes
- Add proper text when deliveries are 0

Jacobjeevan committed:
add debounced to purchase order suppliers

Jacobjeevan committed:
mv vendor out in purchase order

Jacobjeevan committed:
added purchase order by vendor view

abhimanyurajeesh committed:
Update PurchaseOrders component and localization

- Added new translation for "ordered" in en.json.
- Updated TABS_CONFIG to include "ordered" status and modified existing statuses for clarity.
- Refined tab trigger styles for improved UI consistency and responsiveness.

yash-learner committed:
Reuse the SRD for delivery see details in to_dispatch

and move the SRD to new directory

yash-learner committed:
Pass the current params in naviagate urls

abhimanyurajeesh committed:
Refactor ToDispatchSupplyRequestTable component

- Moved TableSkeleton import to the correct location.
- Updated Table import path to use the Common directory for better modularization.

yash-learner committed:
Remove condition adn reset form

- Move to mutateAsync

yash-learner committed:
Check for unit from supply_inventory_item as well

yash-learner committed:
Use Promise.allSettled for better error handling

yash-learner committed:
Add confirm dialog

so that user can still mark supply request as completd in case they miss that or proceed as it is

yash-learner committed:
Fix the status values in receive form and button disable state

yash-learner committed:
Let's reuse the confirmactiondailog and fix some bugs

yash-learner committed:
Remove entered in error on receive side and render layout based on delivery status

yash-learner committed:
Add option to update an abandoned item to complete

yash-learner committed:
TIny changes

- add key
- Update the dialog title

yash-learner committed:
Update ConfirmActionDialog

- Add disabled and hideCancel prop

yash-learner committed:
Reuse the ConfirmActionDiaog inside SRD

- Upda the ReceiveItem CAD to use new props

rithviknishad committed:
Merge remote-tracking branch 'upstream/labs-v2' into inventory-internal-transfers

yash-learner committed:
Add highlight animation to delivery item

- This will be useful when there are multiple deliveries for a SR

yash-learner committed:
Add explicit mark as fully dispacthed action

- refactor and tiny chnages

yash-learner committed:
Try adding all the missing translation keys

yash-learner committed:
Disable the confirm dispatch when any of the item fields are empty

yash-learner committed:
Change the order of the form items

yash-learner committed:
Fix the query to get internal deliveries correctly

abhimanyurajeesh committed:
Merge remote-tracking branch 'upstream/labs-v2' into inventory-internal-transfers

abhimanyurajeesh committed:
Merge remote-tracking branch 'upstream/labs-v2' into inventory-internal-transfers

abhimanyurajeesh committed:
feat: Add processed status color to supply request and update type mapping for Badge component

Jacobjeevan committed:
incoming deliveries/inward entry UI

abhimanyurajeesh committed:
refactor: Replace hardcoded badge colors with centralized constants for supply request status and priority across components

abhimanyurajeesh committed:
Merge remote-tracking branch 'upstream/labs-v2' into inventory-internal-transfers

Jacobjeevan committed:
receive stock UI changes

abhimanyurajeesh committed:
Merge remote-tracking branch 'upstream/labs-v2' into inventory-internal-transfers

abhimanyurajeesh committed:
Merge remote-tracking branch 'upstream/labs-v2' into inventory-internal-transfers

abhimanyurajeesh committed:
Enhance ProductKnowledgeSelect component to accept a placeholder prop and update its usage across multiple tables for improved search functionality.

Jacobjeevan committed:
add charge item info to receive stock table, support for delete item

Jacobjeevan committed:
added support for misc item receival + type changes/minor cleanup

abhimanyurajeesh committed:
Implement OrgSelect component for supplier selection in IncomingDeliveries and PurchaseOrders; add new translations for organization search functionality.

abhimanyurajeesh committed:
Merge remote-tracking branch 'upstream/labs-v2' into inventory-internal-transfers

Jacobjeevan committed:
fixing issues in supply req/delivery flow

yash-learner committed:
Add link to request page on supply delivery detail and some back url fixes

yash-learner committed:
Add the mark as fully received action and tiny fixes

- Show when there is atleast one delivery completed and status is active or processed
- Some fixes in passing params

yash-learner committed:
Fix the params and clearin where required

yash-learner committed:
Fix the RadioItem selection via label click

yash-learner committed:
Tiny refactor

yash-learner committed:
Add draft status tab and rename ordered tab to processed

yash-learner committed:
WIP: create a common form for both internal and external supply request create and edit

yash-learner committed:
Use the common SRF in purchase orders as well

yash-learner committed:
Reuse the SR detail component from internal transfer for external SR detail view

and do not render the sidebar in some screens

yash-learner committed:
Show the add another item in edit mode as well

yash-learner committed:
Fix the field alignment

yash-learner committed:
Improve i18n a bit

yash-learner committed:
Remove the old supply request and delivery tabs

- Todo: later remove remaining related files as well

abhimanyurajeesh committed:
Merge remote-tracking branch 'upstream/labs-v2' into inventory-internal-transfers

abhimanyurajeesh commented on src/Routers/AppRouter.tsx:
```diff
@@ -35,6 +35,12 @@ const PATHS_WITHOUT_SIDEBAR = [
   "/session-expired",
   // Pattern matches (using regex)
   /^\/facility\/[^/]+\/services_requests\/[^/]+$/,
+  /^\/facility\/[^/]+\/locations\/[^/]+\/internal_transfers\/to_receive\/[^/]+$/,
+  /^\/facility\/[^/]+\/locations\/[^/]+\/internal_transfers\/to_dispatch\/[^/]+$/,
+  /^\/facility\/[^/]+\/locations\/[^/]+\/internal_transfers\/requests\/[^/]+$/,
+  /^\/facility\/[^/]+\/locations\/[^/]+\/internal_transfers\/requests\/[^/]+\/edit$/,
+  /^\/facility\/[^/]+\/locations\/[^/]+\/external_supply\/purchase_orders\/new$/,
+  /^\/facility\/[^/]+\/locations\/[^/]+\/external_supply\/purchase_orders\/[^/]+\/edit$/,
```
we could do an or on the regex and simplify it right?

yash-learner replied:
It is a nice way to reduce the code a bit, but I feel it makes it harder to read, and we can break it at some point

yash-learner committed:
Update the RecieveItem component from internal transfers and reuse it inside the external supply delivert view

yash-learner committed:
Clean up old approve flow in external

yash-learner committed:
Fix redirect and render a page without the sidebar

yash-learner committed:
Fix some fallback values and tiny things

Jacobjeevan committed:
receive stock, healthcare service bug fixes

yash-learner committed:
Merge branch 'labs-v2' into inventory-internal-transfers
